processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -240,9 +255,10 @@ impl<E: FieldElement<BaseField = Felt>> AuxColumnBuilder<E> for BusColumnBuilder
         let op_code = op_code_felt.as_int() as u8;
 
         match op_code {
-            OPCODE_JOIN | OPCODE_SPLIT | OPCODE_LOOP | OPCODE_DYN | OPCODE_CALL => {
+            OPCODE_JOIN | OPCODE_SPLIT | OPCODE_LOOP | OPCODE_CALL => {
                 build_control_block_request(main_trace, op_code_felt, alphas, row)
             },
+            OPCODE_DYN => build_dyn_block_request(main_trace, alphas, row),
```

Comment:
As discussed offline, I think the handling of the `DYN` operation here was correct. Which means there is a bug somewhere else - probably [here](https://github.com/0xPolygonMiden/miden-vm/blob/next/processor/src/decoder/mod.rs#L540) where we shouldn't be putting `dyn_hash` into decoder registers (or at least that would be consistent with the current [docs](https://0xpolygonmiden.github.io/miden-vm/design/decoder/main.html#dyn-block-decoding)).

Reply:
Pushed a commit that fixes this as explained here. Basically `DYN` emits a request to hash `[ZERO; 8]` using the `DYN` domain, which is verified on `END`. The block hash table is responsible for ensuring that the correct hash is being executed: the callee hash is pushed on the table on `DYN`, and removed on the appropriate `END` (when the callee terminates).

Reply:
It seems like `DYN` could not issue any hash request, and also not have an associated `END`, since we're only hashing dummy values? So `DYN` would still populate the block hash table, but have no other effect on the trace.

Reply:
> It seems like `DYN` could not issue any hash request, and also not have an associated `END`, since we're only hashing dummy values? So `DYN` would still populate the block hash table, but have no other effect on the trace.

Yeah - I think that's correct.